internal/usecases/create_kubeconfig.go

```diff
@@ -56,15 +58,16 @@ func (u *createKubeConfigUseCase) init(ctx context.Context) error {
 	}
 
 	u.conn = conn
-	u.clusterServiceClient = api.NewClusterClient(u.conn)
+	u.clusterAccessClient = api.NewClusterAccessClient(u.conn)
+	u.userClient = api.NewUserClient(u.conn)
 
 	u.kubeConfig = k8s.NewKubeConfig()
 	u.setInitialized()
 
 	return nil
 }
 
-func (u *createKubeConfigUseCase) getNaming(m8ClusterName string, clusterRole mk8s.K8sRole) (clusterName, contextName, nsName, authInfoName string, err error) {
+func (u *createKubeConfigUseCase) getNaming(m8ClusterName string, clusterRole string) (clusterName, contextName, nsName, authInfoName string, err error) {
```

Comment:
what happens if any of the parameter are empty or incomplete or malformed? Is there any validation anywhere?

Reply:
Good point. In fact none of these values should ever be empty or incomplete since they come from the server. Not sure if the client should handle this in any way. No working kubeconfig will ever come out of this if the server side does not provide the right values.

Reply:
i say: let's check for it and be double safe. If it happens for whatever reason we know immediately that the unlikely happened ;-)

Reply:
Actually the method mk8s.GetNamespaceName checks that the input is valid. Still will check that it's not empty.